### PR TITLE
[POC] all: fix deepsource checks

### PR DESCRIPTION
**Pull request by RhinosF1**
### Description
Cleans up a bunch of possible style issues per https://deepsource.io/gh/RhinosF1/sopel/issues/?category=recommended

This is a proof of concept but if you like the feedback then I can go further or it could be configured on your end too.

### Checklist
- [x] I have read [CONTRIBUTING.md](https://github.com/sopel-irc/sopel/blob/master/CONTRIBUTING.md)
- [x] I can and do license this contribution under the EFLv2
- [x] No issues are reported by `make qa` (runs `make quality` and `make test`)
- [x] I have tested the functionality of the things this change touches

### CODE REVIEW

**Comment by Exirel** [comment]
Almost a year now, and I don't think we gave it a proper review. Probably because, as you may have guest, I'm not a huge fan of adding yet another tools with yet another opinions on code style.

I think it's fair to say we can close this. What do you say @dgw ?

File: sopel/config/__init__.py [hunk 1]
```diff
@@ -156,8 +156,7 @@ def homedir(self):
             configured = self.parser.get('core', 'homedir')
         if configured:
             return configured
-        else:
-            return os.path.dirname(self.filename)
+        return os.path.dirname(self.filename)
```
Comment (dgw) [inline sopel/config/__init__.py:159]: But this kind of thing is also good.

File: sopel/config/__init__.py [hunk 2]
```diff
@@ -276,7 +275,7 @@ def define_section(self, name, cls_, validate=True):
         current_name = str(current.__class__)
         new_name = str(cls_)
         if (current is not None and not isinstance(current, self.ConfigSection) and
-                not current_name == new_name):
+                current_name != new_name):
```
Comment (dgw) [inline sopel/config/__init__.py:278]: This is good, too.

**Review by dgw — COMMENTED** [review]
> I'm not a huge fan of adding yet another tools with yet another opinions on code style.

> What do you say @dgw ?

I say what's below in the line notes. **tl;dr:** This tool does have _some_ useful rules, but I can only see documentation on ignoring issues in specific lines. One reason I think flake8 has been our preference for code style is that it lets you globally decide that you don't care about a certain style rule, and in many cases there are multiple rules to choose from (e.g. putting binary operators at the beginning or end of a line).

If there's really no way to customize the style rules applied by DeepSource, then the way forward is to look for flake8 plugins that enforce the DeepSource rules I/we _do_ like and add those (to the wishlist in #1765, probably, so they can be implemented at an opportune moment).

**Pull request closed without merging**